src/jobs/models/jobsManager.ts

```diff
@@ -0,0 +1,266 @@
+
+      const catalogRecord = await this.catalogClient.findRecord(
+        job.metadata.productId as string,
+        highestVersionToString,
+        job.metadata.productType as string
+      );
+
+      const mergedData = this.metadataMerger.merge(catalogRecord?.metadata as LayerMetadata, job.metadata);
+      this.logger.debug({
+        internalId: catalogRecord?.id,
+        metadata: job.metadata,
+        msg: `Updating catalog record ${catalogRecord?.id as string} with new metadata`,
+      });
+      await this.catalogClient.update(catalogRecord?.id as string, mergedData);
+
+      if (job.isSuccessful) {
+        const message = `Updating status of job ${job.id} to be ${OperationStatus.COMPLETED}`;
+        this.logger.info({
+          jobId: job.id,
+          msg: message,
+        });
+        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+        await this.jobManager.updateJobById(job.id, OperationStatus.COMPLETED, 100, undefined, catalogRecord?.id);
+        job.status = OperationStatus.COMPLETED;
+      }
+    }
+  }
+
+  private async handleNewIngestion(job: ICompletedJobs, task: TaskResponse): Promise<void> {
+    if (task.status === OperationStatus.FAILED && job.status !== OperationStatus.FAILED) {
+      await this.abortJobWithStatusFailed(job.id, `Failed to generate tiles`);
+      job.status = OperationStatus.FAILED;
+    } else if (job.isSuccessful) {
+      const layerName = getMapServingLayerName(job.metadata.productId as string, job.metadata.productType as ProductType);
+
+      this.logger.debug({
+        productId: job.metadata.productId,
+        productType: job.metadata.productType,
+        version: job.metadata.productVersion,
+        msg: `[TasksManager][handleNewIngestion] Publishing layer name: "${layerName}" in map services`,
+      });
+
+      await this.publishToMappingServer(job.id, job.metadata, layerName, job.relativePath);
+      const catalogId = await this.publishToCatalog(job.id, job.metadata, layerName);
+      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+      await this.jobManager.updateJobById(job.id, OperationStatus.COMPLETED, 100, undefined, catalogId);
+      job.status = OperationStatus.COMPLETED;
+
+      if (this.shouldSync) {
+        try {
+          await this.syncClient.triggerSync(
+            job.id,
+            job.metadata.productId as string,
+            job.metadata.productVersion as string,
+            job.metadata.productType as ProductType,
+            OperationTypeEnum.ADD,
+            job.relativePath
+          );
+        } catch (err) {
+          const message = `[TasksManager][handleNewIngestion] Failed to trigger sync productId ${job.metadata.productId as string} productVersion ${
+            job.metadata.productVersion as string
+          }. error=${(err as Error).message}`;
+
+          this.logger.error({
+            productId: job.metadata.productId,
```

Comment:
add jobId for all relevant logs (if jobId exists it should be inside the object so we can filter by it in log collector)

Reply:
missed it here, i did add jobId to all logs as possible

Reply:
fixed